GoogleForCreators/web-stories-wp
# Dashboard: Implement new header designs

## Context

Updates dashboard headers to new designs found in figma. 

## Summary

No new functionality. Updates the dashboard header to new designs, improves readability and adds minimum size on dashboard to avoid elements crashing into each other on smaller view ports. 

![header demo](https://user-images.githubusercontent.com/10720454/108416369-d7ff0600-71eb-11eb-823d-71a2a8d4c80f.gif)


## Relevant Technical Choices

- Changes the shared `PageHeading` API so that `showSearch` is false by default, typeahead language is updated to be 'search' based and simplify the styles provided to `children` because now it'll always be centered and we don't have extra ✨  active styles to track. 
- Ripped out the old `typeahead` and `toggleButtonGroup`, as well as tragically out of date template filters that aren't in the new designs and just deadweight in the UI at this point. 
- Adjusted the min-width of dashboard in `PageStructure` to dashboard specific min-width (1098px) 
- Adds disabled color to pills when inactive and disabled. 
- Left aligns empty search w/ placeholder and search icon. No space shift anymore.

## To-do

- Design has made mention of possibly not disabling filters w/ 0 results but shifting the displayed message in the center content of the dashboard to new graphics. For now, no changes here. 
- Eventually it'd be nice to move the `shared/` directory in `views` out since they are pretty much all that will be left from the components, could combine the dashboard components with these since the logic there is minimum. Avoided moving things for the sake of the DIFF here. 

## User-facing changes

Design changes from this: 
![Screen Shot 2021-02-18 at 1 28 42 PM](https://user-images.githubusercontent.com/10720454/108417507-48f2ed80-71ed-11eb-80fc-4a3140a89aef.png)

to this: 
![Screen Shot 2021-02-18 at 1 28 50 PM](https://user-images.githubusercontent.com/10720454/108417521-4e503800-71ed-11eb-8d09-3d71f298aeea.png)

## Testing Instructions

Regression testing: 
- Clicking on filters on 'my stories' (all stories, drafts, published, private) limit the scope of the stories shown in the grid/list view 
- Search functionality still works the same (filtering as you type) 
- Also check that templates and settings views have updated heading style (just the heading is present, no search or filtering) 

New! 
- Try resizing the browser window and make sure that horizontal scroll is enforced when you drop to 1098px. Nothing should overlap anymore. 
- Swap to RTL, see that the header makes sense (this PR should fix #5648)

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #4960 
Fixes #4962
